pennylane/capture/capture_diff.py

```diff
@@ -25,28 +25,31 @@
 
 
 @lru_cache
-def create_non_jvp_primitive():
-    """Create a primitive type ``NonJVPPrimitive``, which binds to JAX's JVPTrace
-    like a standard Python function and otherwise behaves like jax.core.Primitive.
+def create_non_interpreted_prim():
+    """Create a primitive type ``NonInterpPrimitive``, which binds to JAX's JVPTrace
+    and BatchTrace objects like a standard Python function and otherwise behaves like jax.core.Primitive.
```

Comment:
I think it is out of scope for this PR, but we might want to move the `create_non_interpreted_prim` function to a different file and perhaps change the scaffolding structure.